pkgs/by-name/st/structurizr-cli/package.nix

```diff
@@ -0,0 +1,91 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchpatch2,
+  coreutils,
+  findutils,
+  gnused,
+  jre,
+  gradle,
+  makeWrapper,
+  versionCheckHook,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "structurizr-cli";
+  version = "2025.05.28";
+
+  src = fetchFromGitHub {
+    owner = "structurizr";
+    repo = "cli";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-bypCW7fEfSzVQHhb7wzxQUkXnoDb8QHUsPCpHV7pW/w=";
+  };
+
+  patches = [
+    (fetchpatch2 {
+      url = "https://patch-diff.githubusercontent.com/raw/structurizr/cli/pull/175.patch";
+      hash = "sha256-Ut9Rma54vb9UYtvHiCf8LI9i3trlCa45/60sRyKdPnQ=";
+    })
+  ];
+
+  postPatch = ''
+    substituteInPlace src/main/resources/build.properties \
+      --subst-var-by BUILD_NUMBER "${finalAttrs.version}" \
+      --subst-var-by BUILD_DATE "1970-01-01T00:00:00Z"
+  '';
+
+  nativeBuildInputs = [
+    gradle
+    makeWrapper
+  ];
+
+  mitmCache = gradle.fetchDeps {
+    inherit (finalAttrs) pname;
+    data = ./deps.json;
+  };
+
+  __darwinAllowLocalNetworking = true;
+
+  gradleBuildTask = "installDist";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,lib}
+
+    cp -r build/install/structurizr-cli $out/lib/structurizr-cli
```

Comment:
Prefer `install`, remove the `mkdir` step

Reply:
I wasn't sure at first what you meant; I did replace it now with some loop that would try to find all of the relevant files and install each individually, but maybe that's not what you meant?

Reply:
I found the loop a bit complicated to understand as it is, I would use something simpler.
The idea here is not to reduce the repetition of commands, but merely trying to find a good compromise between readability, maintainability and comprehension.

Reply:
> I found the loop a bit complicated to understand as it is
Yes, I would agree with that, I personally also preferred the simple copy command. I did look into some kind of `install` command with a directory, but for some reason, that didn't work (the local build failed).

Do you have an alternative perhaps?

Reply:
Use cp when you need to copy directories, use install when it is a simple file(s).